### PR TITLE
Lift annotation field's priority in make query

### PR DESCRIPTION
**Pull request by elisong**
Lift annotation field's priority in make query when duplicate with model field

## Description

This PR modified `Tortoise.queryset.FieldSelectQuery.add_field_to_select_query()` method by lifting `if field in self.annotations` block before `if field in self.model._meta.fields_db_projection`.

before this PR

```python
Model.filter(id=1).annotate(id=F("id")+1).values("id").sql()
# SELECT `id` `id` FROM `model` where `id`=1

```

after this PR

```python
Model.filter(id=1).annotate(id=F("id")+1).values("id").sql()
# SELECT `id`+1 `id` FROM `model` where `id`=1
```

## Motivation and Context

Sometimes, field name in annotations already exist in model field set, and users want to pick annotated field in `.values(<field>)`.

Suppose a request to calculate aggregation for given metrics: `GET /?metrics=m1,m2,m3`, and expect a response with same field name(frontend don't know backend logic):

before this PR

```python
async def get_average(request):
    metrics = request.query_params.get('metrics').split(',')
    data = await Model.all().annotate(
            **{f"some_{m}_chareacter": AVG(m) for m in metrics}
        ).values(
            **{m: f"some_{m}_chareacter" for m in metrics}
        )
    return JSONResponse(data)
```

after this PR

```python
async def get_average(request):
    metrics = request.query_params.get('metrics').split(',')
    data = await Model.all().annotate(**{m: AVG(m) for m in metrics}).values(*metrics)
    return JSONResponse(data)
```

## How Has This Been Tested?

add test `test_queryset.TestQueryset.test_annotation_field_priorior_to_model_field()`

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by elisong** [comment]
Does this PR have side effects that i happen to ignore?

File: tortoise/queryset.py [hunk 1]
```diff
@@ -1325,10 +1326,15 @@ async def _execute(self) -> List[Any]:
         ]
         if self.flat:
             func = columns[0][1]
-            flatmap = lambda entry: func(entry["0"])  # noqa
+
```
Comment (long2ice) [inline tortoise/queryset.py:1329]: Why change this?

**Review by long2ice — APPROVED** [review]

File: tortoise/queryset.py (hunk 1 above)
Comment (elisong) [reply tortoise/queryset.py:1329]: It formatted by my editor custom setting, i will re-commit for remove these changes.

**Comment by elisong** [comment]
Removed personal editor autoformat, and updated CHANGELOG.rst.

**Review by long2ice — APPROVED** [review]

**Comment by long2ice** [comment]
Thanks!

**Pull request merged**